packages/core/src/components/DropdownNew/Dropdown.tsx

```diff
@@ -63,44 +70,115 @@ const Dropdown = forwardRef(
     const dropdownMergedRef = useMergeRef(ref, dropdownRef);
 
     const [isFocused, setIsFocused] = useState(false);
+    const [multiSelectedItems, setMultiSelectedItems] = useState<Item[]>([]);
+
+    const handleSingleChange = (selectedItem: Item | null) => {
+      if (!multi && onChange) {
+        (onChange as (item: Item | null) => void)(selectedItem);
+      }
+    };
+
+    const singleDropdown = useDropdownCombobox<Item>(
+      options,
+      autoFocus,
+      isMenuOpen,
+      closeMenuOnSelect,
+      handleSingleChange,
+      onInputChange,
+      onMenuClose,
+      onMenuOpen,
+      onOptionSelect
+    );
+
+    const multiDropdown = useDropdownMultiCombobox<Item>(
+      options,
+      multiSelectedItems,
+      setMultiSelectedItems,
+      autoFocus,
+      multi ? (onChange as (items: Item[]) => void) : undefined,
+      onInputChange,
+      onMenuClose,
+      onMenuOpen,
+      onOptionSelect
+    );
```

Comment:
I think (as a future improvement), that you can benefit from a Context, and pass everything you need to consume in the hooks to there. and in your custom hooks consume the Context.